### ISSUE TITLE
Path library - add support for expandVars

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#6009]
Back in late 2014, we started undertaking the implementation of the FileSystem and Path libraries, with input from the user community ([start of thread](https://sourceforge.net/p/chapel/mailman/message/32800541/), [mail with updated proposal after feedback](https://sourceforge.net/p/chapel/mailman/message/32902710/)) Much progress has been made in getting the support onto master, but 10 functions still remain unimplemented due to the youth of our string implementation at the time.  With our string implementation in a good state, these functions should be returned to and implemented

This issue is for the function `expandVars`.

Here is the expected documentation and declaration:
```chapel
/* Expands any environment variables in the path of the form `$<name>` or
   `${<name>}` into their values.  If <name> does not exist, they are left
   in place. Returns the path which includes these expansions.

   :arg path: a string representation of a path, which may or may not include
                   `$<name>` or `${<name>}`.
   :type path: `string`

   :return: `path`, having replaced all references to environment variables with
                their values
   :rtype: `string`
*/
proc expandVars(path: string): string;
```

**Comment by lydia-duncan** [comment]
Tests will need to be written for this feature

**Comment by lydia-duncan** [comment]
Resolved by #8617
